### PR TITLE
Proxy agent: panic after retries, broadcast abort, minor fixes

### LINKED ISSUE #400: Istio-proxy should exit when envoy cannot be started or restart fails
This is to recover by letting kubelet restart the istio-proxy pod.

### PR DESCRIPTION
**Pull request by kyessenov**
fixes #400

### CODE REVIEW

File: proxy/agent.go [hunk 1]
```diff
@@ -99,28 +97,37 @@ type Retry struct {
...
+	// Run command for a config, epoch, and abort channel
 	Run func(interface{}, int, <-chan error) error
 
-	// Cleanup command
+	// Cleanup command for an epoch
 	Cleanup func(int)
+
+	// Panic command is invoked when all retries to start the proxy fail
+	// just before the agent terminating
+	Panic func()
```
Comment (andraxylia) [inline proxy/agent.go]: Can we give a different name than Panic? It seems like a very controller shut down of all epochs.

Comment (kyessenov) [inline proxy/agent.go]: I think we need to wave a red flag here since it's a very bad error case. Any proposal?

Comment (andraxylia) [inline proxy/agent.go]: KillThemAll

Comment (kyessenov) [inline proxy/agent.go]: Ah, but we don't kill all proxies here. We just exit the process and let panic also panic. I guess we should kill all proxies before panicking?

Comment (ayj) [inline proxy/agent.go]: Can you pass the current (bad?) configuration to Panic so it can be logged? Would be useful to debug / triage later.

Comment (andraxylia) [inline proxy/agent.go]: +1

Comment (kyessenov) [inline proxy/agent.go]: added as an argument. the config is large and is printed ten times before panic, maybe let's not do that in sidecar yet.

Comment (ayj) [inline proxy/agent.go]: 10 times? Shouldn't Panic() only be called once?

Comment (kyessenov) [inline proxy/agent.go]: There are 10 retries before panic. Each retry prints out the whole config to the log.

**Pull request merged**